src/compile/tests.rs

```diff
@@ -136,3 +136,23 @@ fn test_workspace_project2() {
 
     assert_display_snapshot!(cargo, @"cargo build --package=project2 --lib --target-dir=/celPool/celData/Works/projects/cargo-leptos/examples/workspace/target/front --target=wasm32-unknown-unknown --no-default-features --features=hydrate");
 }
+
+#[test]
+fn test_extra_cargo_args() {
+    let cli = Opts {
+        lib_cargo_args: Some(vec!["-j".into(), "8".into()]),
+        bin_cargo_args: Some(vec!["-j".into(), "16".into()]),
+        ..dev_opts()
+    };
+    let conf = Config::test_load(cli, "examples", "examples/project/Cargo.toml", true);
+
+    let mut command = Command::new("cargo");
+    let (_, cargo) = build_cargo_server_cmd("build", &conf.projects[0], &mut command);
+
+    assert_display_snapshot!(cargo, @"cargo build --package=example --bin=example --no-default-features --features=ssr -j 16");
```

Comment:
So, this test is still probably somewhat useful because the functionality wasn't tested before at all, but it... doesn't actually work as a regression test for this issue. The problem is joining everything to a single string for the snapshot masks the problem right now.

One way to fix this would be to wrap arguments that contain spaces into quotes when building this string, another would be to make `build_cargo_server_cmd` and `build_cargo_front_cmd` return a `Vec<String>` instead of a `String` as the second element in the tuple, if that's practical.

Should I do any of these in this PR?

Reply:
Actually I just went ahead and made the rendered cargo command wrap arguments containing spaces with quotes, see the second commit.

Now, if the bug is re-introduced, the snapshots will fail:

```
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: extra_cargo_args
Source: src/compile/tests.rs:152
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Expression: cargo
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
    0       │-cargo build --package=example --bin=example --no-default-features --features=ssr -j 16
          0 │+cargo build --package=example --bin=example --no-default-features --features=ssr '-j 16'
────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
To update snapshots run `cargo insta review`
Stopped on the first failure. Run `cargo insta test` to run all snapshots.
```

It will also make the user facing output more correct in some cases since this also seems to be used in logs.